File: img_load/src/main/java/com/ad/img_load/pickerview/wheelview/timer/MessageHandler.java
package com.ad.img_load.pickerview.wheelview.timer;

import android.os.Handler;
import android.os.Message;

import androidx.annotation.NonNull;

import com.ad.img_load.pickerview.wheelview.view.WheelView;

import java.lang.ref.WeakReference;


/**
 * Handler 消息类
 *
 * @author 小嵩
 * date: 2017-12-23 23:20:44
 */
public final class MessageHandler extends Handler {
    public static final int WHAT_INVALIDATE_LOOP_VIEW = 1000;
    public static final int WHAT_SMOOTH_SCROLL = 2000;
    public static final int WHAT_ITEM_SELECTED = 3000;

    private final WeakReference<WheelView> wheelView;

    public MessageHandler(WheelView wheelView) {
        this.wheelView = new WeakReference(wheelView);
    }

    @Override
    public final void handleMessage(@NonNull Message msg) {
        WheelView view = wheelView.get();
        if (view == null) return;
        switch (msg.what) {
            case WHAT_INVALIDATE_LOOP_VIEW:
                view.invalidate();
                break;

            case WHAT_SMOOTH_SCROLL:
                view.smoothScroll(WheelView.ACTION.FLING);
                break;

            case WHAT_ITEM_SELECTED:
                view.onItemSelected();
                break;
        }
    }

}

File: Widgets/src/main/java/com/colin/library/android/widget/picker/wheel/MessageHandler.kt
package com.colin.library.android.widget.picker.wheel

import android.os.Handler
import android.os.Looper
import android.os.Message
import java.lang.ref.WeakReference

/**
 * Author:ColinLu
 * E-mail:945919945@qq.com
 * Create:2025-05-02 09:24
 *
 * Des   :MessageHandler
 */
class MessageHandler(wheelView: WheelView?) : Handler(Looper.getMainLooper()) {
    private val wheelView: WeakReference<WheelView?> = WeakReference<WheelView?>(wheelView)

    override fun handleMessage(msg: Message) {
        val view: WheelView? = wheelView.get()
        if (view == null) return
        when (msg.what) {
            WHAT_INVALIDATE_LOOP_VIEW -> view.invalidate()
            WHAT_SMOOTH_SCROLL -> view.smoothScroll(WheelView.ACTION.FLING)
            WHAT_ITEM_SELECTED -> view.onItemSelected()
        }
    }

    companion object {
        const val WHAT_INVALIDATE_LOOP_VIEW: Int = 1000
        const val WHAT_SMOOTH_SCROLL: Int = 2000
        const val WHAT_ITEM_SELECTED: Int = 3000
    }
}
